### PR TITLE
Optimizing Rational

### PR DESCRIPTION
**Pull request by tadd**
This PR is for detailed and additional information about https://bugs.ruby-lang.org/issues/12484.

### CODE REVIEW

File: rational.c [hunk 1]
```diff
@@ -420,8 +402,6 @@ nurat_s_alloc(VALUE klass)
     return nurat_s_new_internal(klass, ZERO, ONE);
 }
 
-#define rb_raise_zerodiv() rb_raise(rb_eZeroDivError, "divided by 0")
-
 #if 0
```
Comment (mrkn) [inline rational.c]: Why not remove this disabled range?

Comment (mrkn) [inline rational.c]: Why not remove this disabled range?

Comment (tadd) [inline rational.c]: OK, I will remove this after the PR merged (issue closed).

**Comment by tadd** [comment]
Rebased and resolved conflict.

**Comment by mrkn** [comment]
I've merged all of the commits in this PR into trunk with some modifications.
The commits exist between 8a847fc046f3804fe0f687b8ff4e95db6ae9bf50 and 379c4cdd16a0e82c19d8ff66da8af0f0b9c334fa.

**Comment by tadd** [comment]
Thank you very much!

File: rational.c (hunk 1 above)
Comment (tadd) [reply rational.c]: I remembered this just today, but I confirmed this was fixed by c225d3c9b52e by @mrkn . Thank you!

**Pull request closed without merging**